lib/TPP/TestForToForAllRewrite.cpp

```diff
@@ -0,0 +1,43 @@
+//===- TestForToForAllRewrite.cpp - ---------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/TransformUtils.h"
+#include "mlir/IR/FunctionInterfaces.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+
+namespace {
+// This is a test pass to check the scf.for to scf.forall rewrite.
+struct TestForToForAllRewrite
```

Comment:
I'm not sure I get the point of this test pass...

Reply:
It tests the rewrite in isolation, similar to what we did for the matchers.

Reply:
The matchers do something, this one just runs greedly. Wouldn't it be the same as running `tpp-opt` with the pass name?

Reply:
It is a pass in tpp-opt, but it is not exposed as we do with the other passes. I chose this because I am only interested in exposing the rewrite pattern that other passes like tile-and-fuse or rewrite-to-brgemm can use. I don't see this as a fully standalone pass, and we don't need to add another pass to the default pipeline. Anyway, I am open to suggestions/improvements.

Reply:
I see. I'm not a fan of this approach, but I don't have a better idea right now. We should think about how we want to continue here to avoid the boilerplate this adds to both code and tests. For now, this is fine.